x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlResolveFieldsResponse.java

```diff
@@ -12,57 +12,78 @@
 import org.elasticsearch.action.fieldcaps.FieldCapabilitiesResponse;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
-import org.elasticsearch.core.Nullable;
 
 import java.io.IOException;
 
 public class EsqlResolveFieldsResponse extends ActionResponse {
     public static final TransportVersion RESOLVE_FIELDS_RESPONSE_CREATED_TV = TransportVersion.fromName(
         "esql_resolve_fields_response_created"
     );
+    public static final TransportVersion RESOLVE_FIELDS_RESPONSE_REMOVED_MIN_TV = TransportVersion.fromName(
+        "esql_resolve_fields_response_removed_min_tv"
+    );
 
     private final FieldCapabilitiesResponse caps;
-    private final TransportVersion minTransportVersion;
 
-    public EsqlResolveFieldsResponse(FieldCapabilitiesResponse caps, TransportVersion minTransportVersion) {
+    public EsqlResolveFieldsResponse(FieldCapabilitiesResponse caps) {
         this.caps = caps;
-        this.minTransportVersion = minTransportVersion;
     }
 
     public EsqlResolveFieldsResponse(StreamInput in) throws IOException {
```

Comment:
NIT: lets replace entire constructor with static method that reads the entire response from StreamInput.